keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ExceptionHandler.kt

```diff
@@ -86,14 +88,28 @@ class ExceptionHandler(
   @ResponseStatus(FORBIDDEN)
   fun onAccessDenied(e: AccessDeniedException): ApiError {
     log.error(e.message)
-    return ApiError(FORBIDDEN,
+    val authFailure = FiatPermissionEvaluator.getAuthorizationFailure().get()
+    val message = if (authFailure != null) {
+      val user = "User ${AuthenticatedRequest.getSpinnakerUser().orElse("")}"
+      val permission = "${authFailure.authorization.let { if (it == null) "access" else "$it permission" }}"
+      val resourceType = authFailure.resourceType.name.toLowerCase().replace('_', ' ')
+      "Access denied. $user does not have $permission to the ${authFailure.resourceName} $resourceType specified in your request. " +
```

Comment:
If the authorization failure info from Fiat is available, this will now include the username, permission and resource in the error message. For example, "User joe@acme.com does not have READ access to the myteam@acme.com service account specified in your request."